java/maven/src/org/netbeans/modules/maven/classpath/ClassPathProviderImpl.java

```diff
@@ -672,12 +603,14 @@ public ClassPath getActiveClassPath() {
             ClassPath ret = active;
             if (ret == null) {
                 MavenProject mp = proj.getOriginalMavenProject();
-                boolean hasMainModuleDescriptor = mp.getCompileSourceRoots().stream().anyMatch((sourceRoot) -> (new File(sourceRoot, MODULE_INFO_JAVA).exists()));
+                boolean hasMainModuleDescriptor = mp.getCompileSourceRoots().stream()
+                        .anyMatch((sourceRoot) -> Files.exists(Paths.get(sourceRoot, MODULE_INFO_JAVA)));
                 if(hasMainModuleDescriptor) {
                     LOGGER.log(Level.FINER, "TestPathSelector {0} for project {1}: has main module-info.java", new Object [] {logDesc, proj.getProjectDirectory().getPath()}); // NOI18N
                 }
                 
-                boolean hasTestModuleDescriptor = mp.getTestCompileSourceRoots().stream().anyMatch((testSourceRoot) -> (new File(testSourceRoot, MODULE_INFO_JAVA).exists()));
+                boolean hasTestModuleDescriptor = mp.getTestCompileSourceRoots().stream()
+                        .anyMatch((testSourceRoot) -> Files.exists(Paths.get(testSourceRoot, MODULE_INFO_JAVA)));
```

Comment:
(testSourceRoot) -> Files.exists(Paths.get(testSourceRoot, MODULE_INFO_JAVA))
this code kind of asks to be a private method with a meaningful name like "moduleInfoJavaExsists" or something

Reply:
yep thats a good idea since the same snippet is used twice which is another reason to extract it.